File: src/main/kotlin/java_leetcode/CreateTargetArray.java
package java_leetcode;

public class CreateTargetArray {
    private static int[] createTargetArray(int[] nums, int[] index) {
        int size = nums.length;
        int[] res = new int[size];
        for(int i=0; i<size; i++){
            if (index[i] <= i ) {
//                for (int j = size-1; j > index[i]; j--) {
//                    res[j] = res[j-1];
//                }
                if (size - 1 - index[i] >= 0)
                    System.arraycopy(res, index[i], res, index[i] + 1, size - 1 - index[i]);
            }
            res[index[i]] = nums[i];
        }
        return res;
    }
}

File: src/main/kotlin/kotlin_leetcode/CreateTargetArray.kt
package kotlin_leetcode

import java.util.Arrays


fun main() {
    val nums = intArrayOf(0, 1, 2, 3, 4)
    val index = intArrayOf(0, 1, 2, 2, 1)
    val res = CreateTargetArray().createTargetArray(nums, index)
    println(res.contentToString())
}


class CreateTargetArray {
    fun createTargetArray(
        nums: IntArray,
        index: IntArray
    ): IntArray {
        val size = nums.size
        val res = IntArray(size)
        for (i in nums.indices){
            if(index[i] <= i){
                for (j in size-1 downTo index[i]+1 step 1){
                    res[j] = res[j-1]
                }
            }
            res[index[i]] = nums[i]
        }
        return res
    }
}

